### PR TITLE
Renames problem to slug, fixes pexpect hanging-related bugs

### LINKED ISSUE #130: hangs when credential.https://github.com/submit50.username is set but password is not in ssh-agent

### PR DESCRIPTION
**Pull request by dmalan**
In writing up docs, I realized that "problem" doesn't quite work for quizzes, tests, projects, etc. Also fixes #130.

### CODE REVIEW

File: submit50.py [hunk 1]
```diff
@@ -139,18 +140,15 @@ def authenticate(org):
         pass
     authenticate.SOCKET = os.path.join(cache, ORG)
 
-    spawn = pexpect.spawn if sys.version_info < (3, 0) else pexpect.spawnu
```
Comment (dmalan) [inline submit50.py]: Not needed if not logging to `stdout` it seems? http://pexpect.readthedocs.io/en/stable/api/pexpect.html#spawn-class

File: submit50.py [hunk 2]
```diff
@@ -139,18 +140,15 @@ def authenticate(org):
...
-    child = spawn("git -c credential.helper='cache --socket {}' credential fill".format(authenticate.SOCKET))
+    # check cache
+    child = pexpect.spawn("git -c credential.helper='cache --socket {}' credential fill".format(authenticate.SOCKET))
     child.sendline("")
-
-    if child.expect(["Username:", pexpect.EOF]):
-        # Credentials are already cached
+    if child.expect(["Username:", "Password:", pexpect.EOF]) == 2:
```
Comment (dmalan) [inline submit50.py]: `Password:` can happen if `username` is configured in `.gitconfig`.

File: submit50.py [hunk 3]
```diff
@@ -251,11 +249,11 @@ def rlinput(prompt, prefill=""):
 def clear_credentials():
     """Clear git credential cache """
     run("git credential-cache --socket {} exit".format(authenticate.SOCKET))
-    # OSX will sometimes store git credentials in the keyring. Try to remove them
-    try:
-        run("git credential-osxkeychain erase", lines=["host=github.com", "protocol=https", ""])
-    except Error:
-        pass
+    if platform.system() == "Darwin":
```
Comment (dmalan) [inline submit50.py]: Else error shows up on non-Macs in `-v` mode.

File: submit50.py [hunk 4]
```diff
@@ -343,8 +341,8 @@ def run(command, cwd=None, env=None, lines=[], password=None, quiet=False, timeo
 
     # wait for prompt, send password
     if password:
-        res = child.expect(["Password for '.*': ", pexpect.EOF])
-        if res == 0:
+        i = child.expect(["Password for '.*': ", pexpect.EOF])
+        if i == 0:
```
Comment (dmalan) [inline submit50.py]: Consistency. Also more clear an index.

File: submit50.py (hunk 2 above)
Comment (crossroads1112) [inline submit50.py]: Nice catch!

File: submit50.py (hunk 1 above)
Comment (crossroads1112) [inline submit50.py]: Per [this section[(http://pexpect.readthedocs.io/en/stable/api/pexpect.html#handling-unicode), if `spawn` is used in python3, the arguments to `expect` are expected to be bytestrings as well. When run under python3 with credentials cached I get  `TypeError: cannot use a string pattern on a bytes-like object`

**Pull request merged**